File: Data Structures & Algorithms/house-robber/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n=len(nums)
        dp=[0]*(n)
        dp[n-1]=nums[n-1]
        for i in range (n-2,-1,-1):
            if i!= n-2:
                dp[i]=max(nums[i]+dp[i+2],dp[i+1])
            else:
                dp[i]=max(nums[i],dp[i+1])
        return dp[0]

File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if(nums.size() <= 1)
            return true;
        if(nums[0]==0)
            return false;
        int maxIndex=nums[0];
        for(int i=1;i<nums.size();i++){
            if(maxIndex>=nums.size()-1)
                return true;
            if (i > maxIndex)
                return false;
            maxIndex=max(maxIndex,nums[i]+i);
        }
        return maxIndex>=(nums.size()-1);
    }
};
